ui/apps/platform/src/Containers/SystemConfig/Form/SystemConfigForm.tsx

```diff
@@ -111,6 +111,13 @@ const SystemConfigForm = ({
                         if (data.publicConfig?.telemetry?.enabled && telemetryConfig.storageKeyV1) {
                             initializeSegment(telemetryConfig.storageKeyV1, telemetryConfig.userId);
                         }
+
+                        const isTelemetryEnabledPrev = publicConfig.telemetry?.enabled;
+                        const isTelemetryEnabledCurr = data.publicConfig?.telemetry?.enabled;
+                        if (isTelemetryEnabledPrev && !isTelemetryEnabledCurr) {
+                            window.location.reload();
+                        }
```

Comment:
I haven't deployed and tested, but on first glance, it looks like this should come _after_ the code that follows it. Otherwise, the settings might not be cleaned up property.

Reply:
I believe if the page reloads, the below shouldn't matter since it's state and will be reset to default values regardless. However, your suggestion is definitely safer regardless.